### ISSUE TITLE
no-unused-expression incorrectly fails with optional-chaining (babel-eslint)

### ISSUE THREAD

**Issue by marudor** [eslint/eslint#10939]
**Tell us about your environment**

* **ESLint Version:** 5.6.1
* **Node Version:** 8.12.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESlint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  parser: 'babel-eslint',
  rules: {
    'no-unused-expressions': 1,
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function testFn() {}
testFn?.();
```

```bash
npx eslint <filename>
```

**What did you expect to happen?**
Expect no error as I used testFn

**What actually happened? Please include the actual, raw output from ESLint.**
```
2:1  error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
```

Happens because babylon reports OptionalChaining as type "OptionalCallExpression" since 8.2.2. With 8.2.1 it was just "CallExpression".
Both report the node as optional as well.

**Comment by psychobolt** [comment]
Hi I have this issue. ESLint 5.7. I see the fix and the issue is closed but it isn't merged yet into master. Is there a reason?

**Comment by marudor** [comment]
Please use https://github.com/babel/eslint-plugin-babel and there ruleset instead of the default one for babel-eslint.
